lib/reporting/reports/sales_tax/sales_tax_totals_by_producer.rb

```diff
@@ -0,0 +1,169 @@
+          # 2. we split each key that has more than tax_rate
+          report_line_items.list.group_by do |line_item|
+            [
+              line_item.tax_rates.map(&:id),
+              line_item.supplier_id,
+              line_item.order.distributor_id,
+              line_item.order.order_cycle_id
+            ]
+          end.flat_map do |k, v|
+            k.first.map do |tax_rate_id|
+              {
+                [tax_rate_id] + k[1..] => v
+              }
+            end
+          end
+        end
+
+        def columns
+          {
+            distributor: :distributor,
+            distributor_tax_status: :distributor_tax_status,
+            producer: :producer,
+            producer_tax_status: :producer_tax_status,
+            order_cycle: :order_cycle,
+            tax_category: :tax_category,
+            tax_rate_name: :tax_rate_name,
+            tax_rate: :tax_rate_amount,
+            total_excl_tax: :total_excl_tax,
+            tax: :tax,
+            total_incl_tax: :total_incl_tax
+          }
+        end
+
+        def rules
+          [
+            {
+              group_by: :distributor,
+            },
+            {
+              group_by: :producer,
+            },
+            {
+              group_by: :order_cycle,
+              summary_row: proc do |_key, items, _rows|
+                line_items = items.flat_map(&:values).flatten.uniq
+                total_excl_tax = line_items.sum(&:amount) - line_items.sum(&:included_tax)
+                tax = line_items.map do |line_item|
+                  line_item.adjustments.eligible.tax.sum(&:amount)
+                end.sum
+                {
+                  total_excl_tax: total_excl_tax,
+                  tax: tax,
+                  total_incl_tax: total_excl_tax + tax
+                }
+              end
+            }
+          ]
+        end
+
+        protected
+
+        def distributor(query_result_row)
+          first_line_item(query_result_row).order.distributor&.name
+        end
+
+        def distributor_tax_status(query_result_row)
+          first_line_item(query_result_row).order.distributor&.charges_sales_tax
+        end
+
+        def producer(query_result_row)
+          first_line_item(query_result_row).supplier.name
+        end
+
+        def producer_tax_status(query_result_row)
+          first_line_item(query_result_row).supplier.charges_sales_tax
+        end
+
+        def order_cycle(query_result_row)
+          first_line_item(query_result_row).order_cycle&.name
+        end
+
+        def tax_category(query_result_row)
+          first_line_item(query_result_row).tax_category&.name
+        end
+
+        def tax_rate_name(query_result_row)
+          tax_rate(query_result_row)&.name
+        end
+
+        def tax_rate_amount(query_result_row)
+          amount = tax_rate(query_result_row)&.amount
+          "#{amount * 100} %"
```

Comment:
I wonder if this is the same in every language. And if we need to distinguish between machine-readable CSV and other reports.

Reply:
I'll use ` I18n.t`

Reply:
what do you think about updating this?
https://github.com/openfoodfoundation/openfoodnetwork/blob/eb663af292a94e2360e0c58ca6a51a6d8531a50b/lib/reporting/report_row_builder.rb#L76-L101

Formatting will only happen when the format is not `:json` or `:csv`.
If the user selects a different format from those two, we use `I18n` after multiplying by 100.

Reply:
That's a great idea. It makes it easier to calculate percentages with CSV and JSON but makes it readable in all other formats. Brilliant.

Reply:
done